src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,28 +111,38 @@ data CCurrency
     deriving (Show, Read, Generic)
 
 -- | Client hash
-newtype CHash = CHash Text deriving (Show, Eq, Generic, Buildable)
+newtype CHash = CHash Text
+    deriving (Show, Eq, Generic, Buildable)
 
 instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash
```

Comment:
I'm going to leave a comment then. I like this sentence on SO - "The motivation behind using phantom types is to specialize the return type of data constructors". It's true that they are both `CAddress`, but I don't get why are we even using phantom types since it seems to me that:
- either a single `CAddress` is enough for both cases - `CAddress` is the same, we don't need to change anything
- there are two different types - in which case we could make the thing more explicit by using sum types? If we are using phantom types, we could at least restrict the types to just the two cases - http://stackoverflow.com/questions/28247543/motivation-behind-phantom-types